libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts

```diff
@@ -0,0 +1,231 @@
+      errorCohortList.push(errorCohortEntry);
+    }
+  }
+  return [errorCohortList, errorStrings];
+}
+
+function translatePreBuiltCohortFilterForTarget(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  jointDataset: JointDataset,
+  cohortColumnName: CohortColumnNames
+): IFilter {
+  let filterColumnName = JointDataset.PredictedYLabel;
+  if (cohortColumnName === CohortColumnNames.TrueY) {
+    filterColumnName = JointDataset.TrueYLabel;
+  }
+  if (
+    jointDataset.getModelType() === ModelTypes.Multiclass ||
+    jointDataset.getModelType() === ModelTypes.Binary
+  ) {
+    const modelClasses = jointDataset.getModelClasses();
+    const index: number[] = [];
+    for (const modelClass of preBuiltCohortFilter.arg) {
+      const indexModelClass = modelClasses.indexOf(modelClass);
+
+      if (indexModelClass !== -1) {
+        index.push(indexModelClass);
+      }
+    }
+
+    index.sort((a, b) => a - b);
+
+    const filter: IFilter = {
+      arg: index,
+      column: filterColumnName,
+      method: preBuiltCohortFilter.method
+    } as IFilter;
+    return filter;
+  }
+  const filter: IFilter = {
+    arg: preBuiltCohortFilter.arg,
+    column: filterColumnName,
+    method: preBuiltCohortFilter.method
+  } as IFilter;
+  return filter;
+}
+
+function translatePreBuiltCohortFilterForClassificationOutcome(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  jointDataset: JointDataset
+): IFilter {
+  const index: number[] = [];
+  if (JointDataset.ClassificationError in jointDataset.metaDict) {
+    const allowedCalssificationErrorValues =
+      jointDataset.metaDict[JointDataset.ClassificationError]
+        .sortedCategoricalValues;
+
+    if (allowedCalssificationErrorValues !== undefined) {
+      for (const classificationError of preBuiltCohortFilter.arg) {
+        const indexclassificationError =
+          allowedCalssificationErrorValues.indexOf(classificationError);
+
+        if (indexclassificationError !== -1) {
+          index.push(indexclassificationError);
+        }
+      }
+    }
+  }
+  index.sort((a, b) => a - b);
+  const filter: IFilter = {
+    arg: index,
+    column: JointDataset.ClassificationError,
+    method: preBuiltCohortFilter.method
+  } as IFilter;
+  return filter;
+}
+
+function translatePreBuiltCohortFilterForDataset(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  jointDataset: JointDataset
+): [IFilter | undefined, string | undefined] {
+  let jointDatasetFeatureName = undefined;
+  let userDatasetFeatureName = undefined;
+  for (jointDatasetFeatureName in jointDataset.metaDict) {
+    if (
+      jointDataset.metaDict[jointDatasetFeatureName].abbridgedLabel ===
+      preBuiltCohortFilter.column
+    ) {
+      userDatasetFeatureName =
+        jointDataset.metaDict[jointDatasetFeatureName].abbridgedLabel;
+      break;
+    }
+  }
+
+  if (
+    jointDatasetFeatureName === undefined ||
+    userDatasetFeatureName === undefined
+  ) {
+    return [undefined, "Feature name not found in the dataset"];
```

Comment:
this looks like user facing message, needs to be localized

Reply:
Done